integration/test/app/kubeconfig/kubeconfig_test.go

```diff
@@ -59,173 +56,39 @@ func TestAppWithKubeconfig(t *testing.T) {
+		bytes, err := clientcmd.Write(*c)
 		if err != nil {
 			t.Fatalf("expected nil got %#v", err)
 		}
-	}
-
-	{
-		config.Logger.Debugf(ctx, "creating kubeconfig secret")
-
-		_, err = config.K8sClients.K8sClient().CoreV1().Secrets(key.Namespace()).Create(ctx, &corev1.Secret{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      kubeConfigName,
-				Namespace: key.Namespace(),
-			},
-			Data: map[string][]byte{
-				"kubeConfig": bytes,
-			},
-		}, metav1.CreateOptions{})
-		if err != nil {
-			t.Fatalf("expected nil got %#v", err)
-		}
-
-		config.Logger.Debugf(ctx, "created kubeconfig secret")
-	}
-
-	{
-		config.Logger.Debugf(ctx, "creating catalog configmap")
-
-		_, err = config.K8sClients.K8sClient().CoreV1().ConfigMaps(key.Namespace()).Create(ctx, &corev1.ConfigMap{
-			Data: map[string]string{
-				"values": templates.ChartOperatorValues,
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      catalogConfigMapName,
-				Namespace: key.Namespace(),
-			},
-		}, metav1.CreateOptions{})
-		if err != nil {
-			t.Fatalf("expected nil got %#v", err)
-		}
-
-		config.Logger.Debugf(ctx, "created catalog configmap")
-	}
-
-	{
-		config.Logger.Debugf(ctx, "creating %#q appcatalog cr", key.DefaultCatalogName())
-
-		appCatalogCR := &v1alpha1.AppCatalog{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: key.DefaultCatalogName(),
-				Labels: map[string]string{
-					label.AppOperatorVersion: key.UniqueAppVersion(),
-				},
-			},
-			Spec: v1alpha1.AppCatalogSpec{
-				Config: v1alpha1.AppCatalogSpecConfig{
-					ConfigMap: v1alpha1.AppCatalogSpecConfigConfigMap{
-						Name:      catalogConfigMapName,
-						Namespace: key.Namespace(),
-					},
-				},
-				Description: key.DefaultCatalogName(),
-				Storage: v1alpha1.AppCatalogSpecStorage{
-					Type: "helm",
-					URL:  key.DefaultCatalogStorageURL(),
-				},
-				Title: key.DefaultCatalogName(),
-			},
-		}
-		_, err = config.K8sClients.G8sClient().ApplicationV1alpha1().AppCatalogs().Create(ctx, appCatalogCR, metav1.CreateOptions{})
-		if err != nil {
-			t.Fatalf("expected %#v got %#v", nil, err)
-		}
 
-		config.Logger.Debugf(ctx, "created %#q appcatalog cr", key.DefaultCatalogName())
+		kubeConfig = string(bytes)
 	}
 
 	{
-		config.Logger.Debugf(ctx, "creating chart-operator app CR")
-
-		_, err = config.K8sClients.G8sClient().ApplicationV1alpha1().Apps(key.Namespace()).Create(ctx, &v1alpha1.App{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      key.ChartOperatorName(),
-				Namespace: key.Namespace(),
-				Labels: map[string]string{
-					label.AppOperatorVersion: key.UniqueAppVersion(),
-				},
+		apps := []apptest.App{
+			{
+				// Bootstrap chart-operator in the giantswarm namespace.
+				CatalogName:   key.DefaultCatalogName(),
+				KubeConfig:    kubeConfig,
+				Name:          key.ChartOperatorName(),
+				Namespace:     key.Namespace(),
+				ValuesYAML:    templates.ChartOperatorValues,
+				Version:       key.ChartOperatorVersion(),
+				WaitForDeploy: true,
```

Comment:
For the kubeconfig test we also create a chart-operator app CR to test the bootstrap logic.